inbm/dispatcher-agent/dispatcher/schedule/sqlite_manager.py

```diff
@@ -18,79 +17,293 @@
-        logger.debug("Task added.")
\ No newline at end of file
+            raise DispatcherException(f"Error connecting to Dispatcher Schedule database: {e}")
+    
+    def _create_single_schedule(self, ss: SingleSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        logger.debug(f"Execute -> INSERT INTO single_schedule_manifest(request_id, start_time, end_time) VALUES({ss.request_id}{ss.start_time}{ss.end_time})")
+
+        sql = ''' INSERT INTO single_schedule(request_id, start_time, end_time)
+                                VALUES(?,?,?); '''
+        start_time = None if not ss.start_time else str(ss.start_time)
+        end_time = None if not ss.end_time else str(ss.end_time)
+        try:
+            self._cursor.execute(sql, (ss.request_id, start_time, end_time))
+            schedule_id = self._cursor.lastrowid
+            if not schedule_id:
+                raise DispatcherException("No schedule id was added to the single_schedule table.")
+            
+            logger.debug(f"Added schedule with id: {str(schedule_id)}, request_id: {ss.request_id}, start_time: {str(ss.start_time)}, end_time: {str(ss.end_time)}")
+        
+            # Add the manifests to the manifest table
+            manifest_ids = self._insert_manifest_to_table(ss.manifests)
+            
+            # Add the schedule_id and manifest_id to the single_schedule_manifest table
+            self._insert_single_schedule_manifest_tables(schedule_id, manifest_ids)         
+            self._commit_transaction()
+        except (sqlite3.DatabaseError) as e:
+            self._rollback_transaction()
+            raise DispatcherException(f"Transaction failed: {str(e)}")                       
+
+    def _commit_transaction(self) -> None:
+        if self._conn:
+            self._conn.commit()
+            
+    def _execute_transaction(self, sql: str) -> None:
+        if self._conn:
+            self._conn.execute(sql)
+            
+    def _rollback_transaction(self) -> None:
+        if self._conn:
+            self._conn.rollback()
+
+    def _create_repeated_schedule(self, rs: RepeatedSchedule) -> None:
+        # Add the schedule to the single_schedule table
+        logger.debug(f"Execute -> INSERT INTO repeated_schedule(request_id, cron_duration, cron_minutes, cron_hours, cron_day_month, cron_month, cron_day_week) VALUES({rs.request_id}, {rs.cron_duration}, {rs.cron_minutes}, {rs.cron_hours}, {rs.cron_day_month}, {rs.cron_month}, {rs.cron_day_week})")
+
+        sql = ''' INSERT INTO repeated_schedule(request_id, cron_duration, cron_minutes, cron_hours, cron_day_month, cron_month, cron_day_week)
+                                VALUES(?,?,?,?,?,?,?); '''
+        try:
+            self._cursor.execute(sql, (rs.request_id, 
+                                        rs.cron_duration, 
+                                        rs.cron_minutes, 
+                                        rs.cron_hours, 
```

Comment:
<h3>$\color{red}Defects: Security Vulnerability$</h3>

Use parameterized queries instead of string formatting to prevent SQL injection vulnerabilities. This also makes the code cleaner and more secure.